accord-core/src/main/java/accord/primitives/TxnId.java

```diff
@@ -166,7 +166,7 @@ public static TxnId minForEpoch(long epoch)
         return new TxnId(epochMsb(epoch), 0, Id.NONE);
     }
 
-    private static final Pattern PARSE = Pattern.compile("\\[(?<epoch>[0-9]+),(?<hlc>[0-9]+),(?<flags>[0-9]+)\\([KREWNSXL]\\),(?<node>[0-9]+)]");
+    private static final Pattern PARSE = Pattern.compile("\\[(?<epoch>[0-9]+),(?<hlc>[0-9]+),(?<flags>[0-9]+)\\([KREWNSXL]+\\),(?<node>-?[0-9]+)]");
```

Comment:
bug found in testing.  The property `this == parse(this.toString())` was not valid, and we actually could not parse anything at all!  There is a new property test to make sure this is working and doesn't break again